src/Bus/TestBus.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Zenstruck\Messenger\Test\Bus;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+final class TestBus implements MessageBusInterface
+{
+    private MessageBusInterface $decorated;
+    /** @var array<string, list<Envelope>> */
+    private static array $messages = [];
+
+    // The setting applies to all buses
+    private static bool $enableMessagesCollection = true;
+
+    public function __construct(private string $name, MessageBusInterface $decorated)
+    {
+        $this->decorated = $decorated;
+    }
```

Comment:
please:
- use constructor property promotion for all properties
- initialize the messages list for this bus: `$this->messages[$name] = []`